### ISSUE TITLE
Fix failing tests in MigrationTest

### ISSUE THREAD

**Issue by nfnt** [mesosphere/marathon#4780]
On current `HEAD` (07869fe), "migrate on a legacy database with the same version will do nothing" and "initializes and closes the persistent store when performing a legacy migration" fail.
Stack trace:
```
mesosphere.marathon.MigrationFailedException: Migration Failed: null
      at mesosphere.marathon.storage.migration.Migration$$anonfun$1.applyOrElse(Migration.scala:161)
      at mesosphere.marathon.storage.migration.Migration$$anonfun$1.applyOrElse(Migration.scala:159)
      at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
      at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:216)
      at scala.util.Try$.apply(Try.scala:192)
      at scala.util.Failure.recover(Try.scala:216)
      at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
      at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
      at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
      at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
      at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
      at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
      at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
      Cause: java.lang.NullPointerException
      at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:256)
      at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:251)
      at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
      at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
      at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
      at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
      at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```
```
mesosphere.marathon.MigrationFailedException: Migration Failed: null
      at mesosphere.marathon.storage.migration.Migration$$anonfun$1.applyOrElse(Migration.scala:161)
      at mesosphere.marathon.storage.migration.Migration$$anonfun$1.applyOrElse(Migration.scala:159)
      at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
      at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:216)
      at scala.util.Try$.apply(Try.scala:192)
      at scala.util.Failure.recover(Try.scala:216)
      at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
      at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
      at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
      at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
      at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
      at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
      at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
      Cause: java.lang.NullPointerException
      at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:256)
      at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:251)
      at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
      at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
      at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
      at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
      at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by nfnt** [comment]
Review: https://phabricator.mesosphere.com/D280

**Comment by nfnt** [comment]
Closed with 551458d1ed2a5c169b71981e0588cde66e33aab8
